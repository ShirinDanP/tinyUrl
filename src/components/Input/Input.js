import React from 'react';
import classNames from 'classnames';
import { PropTypes } from 'prop-types';

import styles from './Input.module.css';

export const Input = ({ onChange, placeholder, type, className }) =>
  React.createElement('input', {
    onChange,
    placeholder,
    type,
    className: classNames(styles.input, className),
  });

Input.propTypes = {
  placeholder: PropTypes.string,
  className: PropTypes.string,
  type: PropTypes.string,
  onChange: PropTypes.func,
};
