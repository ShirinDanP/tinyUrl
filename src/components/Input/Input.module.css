.input {
    padding: 10px 8px 10px 8px;
    width: 100%;
    font-size: 13px;
    text-decoration: none;
    color: black;
    background-color: white;
    border: 2px solid black;
    margin-top: 1rem;
}

@media (min-width: 600px) {
    .input {
        width: 50%;
    }
}