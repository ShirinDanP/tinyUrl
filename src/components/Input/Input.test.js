import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import * as enzyme from 'enzyme';
import { Input } from './Input';

configure({ adapter: new Adapter() });

afterEach(cleanup);

describe('Input', () => {
  test('should render with no errors', () => {
    const { container } = render(<Input />);

    expect(container).toMatchSnapshot();
  });

  it('render input with placeholder', () => {
    const props = {
        placeholder: 'test text',
      },
      InputPlaceHolder = enzyme.mount(<Input {...props} />);
    expect(InputPlaceHolder.prop('placeholder')).toEqual('test text');
  });

  it('Test change event', () => {
    const mockCallBack = jest.fn();
    const input = shallow(<Input onChange={mockCallBack} />);
    input.find('input').simulate('change');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
