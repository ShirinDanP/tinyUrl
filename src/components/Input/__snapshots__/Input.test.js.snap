// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render with no errors 1`] = `
<div>
  <input
    class="input"
  />
</div>
`;
